src/Illuminate/Collections/Arr.php

```diff
@@ -594,6 +594,20 @@ public static function isList($array)
         return array_is_list($array);
     }
 
+    /**
+     * Determines if an array is one-dimensional.
+     */
+    public static function isFlat($array): bool
+    {
+        foreach ($array as $entry) {
+            if (is_array($entry)) {
+                return false;
+            }
+        }
+
+        return true;
+    }
```

Comment:
This can be simplified to the following once #56526 is approved:
```suggestion
    public static function isFlat($array): bool
    {
        return self::every($array, fn ($value, $key) => ! is_array($value));
    }
```